pythainlp/text_cleaning/core.py

```diff
@@ -0,0 +1,139 @@
+"""
+    text cleaning in  NLP
+"""
+from nlp import extract_tokens, insert_tokens
+import re
+
+"""
+_____________________________________________________________________________
+| ประเภทข้อมูล    | ตัวอย่างที่ต้องจัดการ                                            |
+| ------------  | -------------------------------------------------------   |
+| อักขระพิเศษ     |               !!!, ..., \~, ^\_^                          |
+| อีโมจิ          |               😊, 😊, ^\_^                               |
+| ตัวเลข         |                   3                                       |
+| URL           | [https://example.com](https://example.com)               |
+| Hashtag       | #เที่ยวไทย, #สนุกสุดๆ                                         |
+| ตัวพิมพ์ใหญ่     | “สวัสดีค่ะ” → “สวัสดีค่ะ” (เปลี่ยนทั้งหมดเป็นพิมพ์เล็ก)                  |
+| คำซ้ำ/คำยืด     | “มากกกกกก” → “มาก” หรือ “ดีมาก”                             |
+| ช่องว่างเกิน      | (เช่นเว้นหลายช่อง)                                            |
+| stopwords     | เช่น “แล้วก็”, “แบบว่า”, “คือ”                                 |
+_____________________________________________________________________________
+
+"""
+
+
+test = """
+    example 1 : 
+
+      สวัสดีค่ะ!!! วันนี้อากาศดีมากๆๆ 😊😊  จะไปเที่ยวกับเพื่อนๆมา 3 คน ที่จังหวัดเชียงใหม่!!!  
+      ลองเข้าไปดูรูปได้นะที่ https://facebook.com #เที่ยวไทย #สนุกสุดๆ  
+      แล้วก็แบบว่า...คือมันดีมากกกกกก~   ขอบคุณค่ะ ^_^  
+    """
+"""
+    example 2 :
+
+    OMG!!! This is sooooo AMAZING!!! 😍😍😍 Check it out: https://wow.com/article/123
+    I can't believe it..... @john_doe was right all along!!! 😂😂😂
+    Go to https://example.com RIGHT NOW!!! You won't regret it. #unbelievable #wowww
+    Sooooo goooooddddddd.... I'm in loooooveeee 💕💕💕
+    Visit www.testsite.org to learn more. This is just... WOW!!! 😱😱
+
+"""
+
+thai_special_chars_unicode = {
```

Comment:
You can also try to use the list defined here:

https://github.com/PyThaiNLP/pythainlp/blob/a06923098c9f57e02bd30140e72f5dea9d5ce50a/pythainlp/__init__.py#L20-L22

(it doesn't include `ฯลฯ` though)